### PR TITLE
AssitantBuilder: Actions components

### PR DESCRIPTION
**Pull request by spolu**
## Description

Split Assistant Builder Action screens into components to prepare for Multi-Actions.

This purely code move, no code was changed.

Next step is to refactor validation of cations so that we pass a `setActionsValid` function so that each action do their own validation (so that we remove custom validation code from the main AssistantBuilder page and remove the `timeFrameError` variable that has nothing to do here.

## Risk

N/A No static variable this time :p

## Deploy Plan

- deploy `front`

### CODE REVIEW

**Review by flvndvd — COMMENTED** [review]
LGTM, left a few small comments. Glad we are giving some love to this react part \o/!

**Review by flvndvd — APPROVED** [review]

File: front/components/assistant_builder/actions/RetrievalAction.tsx [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+import type { TimeframeUnit } from "@dust-tt/types";
+import { useState } from "react";
+
+import AssistantBuilderDataSourceModal from "@app/components/assistant_builder/AssistantBuilderDataSourceModal";
+import DataSourceSelectionSection from "@app/components/assistant_builder/DataSourceSelectionSection";
+import { TIME_FRAME_UNIT_TO_LABEL } from "@app/components/assistant_builder/shared";
+import type { AssistantBuilderState } from "@app/components/assistant_builder/types";
+import { classNames } from "@app/lib/utils";
+
+const deleteDataSource = (
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/components/assistant_builder/actions/RetrievalAction.tsx:12]: Not the same code :)

File: front/components/assistant_builder/ActionScreen.tsx [hunk 2]
```diff
@@ -526,16 +370,12 @@ export default function ActionScreen({
...
-            dataSourceConfigurations={
-              builderState.retrievalConfiguration.dataSourceConfigurations
-            }
-            openDataSourceModal={() => {
-              setShowRetrievalDataSourcesModal(true);
-            }}
-            canAddDataSource={dataSources.length > 0}
-            onDelete={deleteRetrievalDataSource}
+            builderState={builderState}
+            setBuilderState={setBuilderState}
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/components/assistant_builder/ActionScreen.tsx:376]: Def

File: front/components/assistant_builder/actions/DustAppRunAction.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+              Dust Application
+            </a>{" "}
+            of your design before replying. The output of the app (last block)
+            is injected in context for the model to generate an answer. The
+            inputs of the app will be automatically generated from the context
+            of the conversation based on the descriptions you provided in the
+            application's input block dataset schema.
+          </div>
+          <DustAppSelectionSection
+            show={builderState.actionMode === "DUST_APP_RUN"}
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/components/assistant_builder/actions/DustAppRunAction.tsx:107]: Not sure I follow?

**Pull request merged**